bundlewrap/metadata.py

```diff
@@ -24,6 +25,39 @@ class DoNotRunAgain(Exception):
     pass
 
 
+def deepcopy_metadata(obj):
+    """
+    Our own version of deepcopy.copy that doesn't pickle.
+    """
+    if isinstance(obj, METADATA_TYPES):
+        return obj
+    elif isinstance(obj, dict):
+        if isinstance(obj, ATOMIC_TYPES[dict]):
+            new_obj = atomic({})
+        else:
+            new_obj = {}
+        for key, value in obj.items():
+            new_key = copy(key)
```

Comment:
Any particular reason why this no longer checks if `key` is in `METADATA_TYPES`, too? Users can now sneak in unsupported types.

Reply:
They can't because `validate_metadata()` now exists independently and is called from `._set_layer()`.